demo_nodes_cpp_native/CMakeLists.txt

```diff
@@ -20,23 +20,29 @@ find_package(rmw REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(rmw_fastrtps_cpp QUIET)
 
-add_library(talker_native SHARED
-  src/talker.cpp)
-ament_target_dependencies(talker_native
-  "rclcpp"
-  "std_msgs"
-  "rclcpp_components"
-  "rmw_fastrtps_cpp")
-rclcpp_components_register_node(talker_native PLUGIN "demo_nodes_cpp_native::Talker" EXECUTABLE talker)
-
-install(TARGETS
-  talker_native
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib
-  RUNTIME DESTINATION bin)
+include_directories(include)
+
+function(create_executable)
+  add_library(talker_native SHARED
+    src/talker.cpp)
+  target_compile_definitions(talker_native
+    PRIVATE "DEMO_NODES_CPP_NATIVE_BUILDING_DLL")
```

Comment:
This change has nothing to do with the title of the PR.

Reply:
This comment wasn't resolved.

Reply:
@dirk-thomas  changed the PR title

Reply:
The PR is about skipping targets when FastRTPS isn't available. These lines add an unrelated compiler definition which isn't even used in this patch. So please remove both lines from the patch.